### PR TITLE
Collection of minor fixes to BlockStorage

### PR DESCRIPTION
**Pull request by Geod24**

### CODE REVIEW

File: source/agora/node/BlockStorage.d [hunk 1]
```diff
@@ -50,11 +50,12 @@ public interface IBlockStorage
         a Genesis block will be added to the ledger. In this case the calling
         code should treat the block as new and update the set of UTXOs, etc.
 
-        If the blockchain data is corrupt the application will be halted.
+        Returns:
+            `false if the data couldn't be loaded, `true` otherwise.
```
Comment (AndrejMitrovic) [inline source/agora/node/BlockStorage.d]:
```
`false
```

 => 

```
`false`
```

File: source/agora/node/BlockStorage.d [hunk 2]
```diff
@@ -194,24 +208,30 @@ public class BlockStorage : IBlockStorage
...
         if (!this.loadAllIndexes())
-            assert(0);
+            return false;
 
         // Add Genesis if the storage is empty
         if (this.height_idx.length == 0)
         {
             if (!this.saveBlock(GenesisBlock))
-                assert(0);
+                return false;
```
Comment (AndrejMitrovic) [inline source/agora/node/BlockStorage.d:237]: This isn't about loading, but saving. So I think it should stay as `assert(0);`

Comment (AndrejMitrovic) [reply source/agora/node/BlockStorage.d:237]: Although I see you do the same in the calling code.. It's a bit of an interesting case..

Comment (AndrejMitrovic) [reply source/agora/node/BlockStorage.d:237]: Ok based on the commit message it makes sense, let it be like this.

File: source/agora/node/BlockStorage.d [hunk 3]
```diff
@@ -809,22 +808,6 @@ public class BlockStorage : IBlockStorage
...
-    /***************************************************************************
-
-        Remove the index file.
-
-        Params:
-            path = path to the data directory
-
-    ***************************************************************************/
-
-    public static void removeIndexFile (string path)
```
Comment (AndrejMitrovic) [inline source/agora/node/BlockStorage.d]: Was this function unused? It seems out of place in this commit.

**Review by AndrejMitrovic — APPROVED** [review]
LG except the minor comments.

File: source/agora/node/BlockStorage.d (hunk 3 above)
Comment (Geod24) [reply source/agora/node/BlockStorage.d]: Yep it was. And as I was grepping for `buildPath` I came accross this. Want it in another commit ?

Comment (AndrejMitrovic) [reply source/agora/node/BlockStorage.d]: I guess it doesn't matter much if it was no-op.

Comment (Geod24) [reply source/agora/node/BlockStorage.d]: It was actually used in the integration tests. So I reverted the `rm`.

**Comment by Geod24** [comment]
There was a stupid mistake that the integration test caught, fixed now

**Pull request merged**